src/Symfony/Component/ValueExporter/Exporter/AbstractValueExporter.php

```diff
@@ -49,25 +56,26 @@
     /**
      * {@inheritdoc}
      */
-    final public function addFormatters(array $appends, array $prepends = array())
+    final public function addFormatters(array $formatters)
     {
-        foreach ($appends as $append) {
-            if (!$append instanceof $this->formatterInterface) {
-                throw new InvalidFormatterException(get_class($append), self::class, $this->formatterInterface);
-            }
-            if ($append instanceof ExpandedFormatter) {
-                $append->setExporter($this);
+        $this->sortedFormatters = null;
+
+        foreach ($formatters as $formatter) {
+            if (is_array($formatter)) {
+                $priority = (int) $formatter[1];
+                $formatter = $formatter[0];
+            } else {
```

Comment:
You could actually do some unpacking here `list($formatter, $priority) = $formatter;`. I don't really think the int cast is necessary